Mention public API in readme (#682)
wasmx/fizzy: README.md

$ cmake --build .
```

This will build Fizzy as a library. [C API] is provided for embedding Fizzy engine in applications, as well as 
[Rust binding].

A small example of using C API, that loads a wasm binary and executes a function:

```c

> #include <fizzy/fizzy/h>

Reviewer: ```suggestion
#include <fizzy/fizzy.h>
```

$ cmake --build .
```

This will build Fizzy as a library. [C API] is provided for embedding Fizzy engine in applications.

> A small example of using C API, that loads a wasm binary and executes a function:

Reviewer: ```suggestion
A small example of using the C API, which loads a wasm binary and executes a function named `main`:
```

```c
#include <fizzy/fizzy.h>

bool execute_main(const uint8_t* wasm_binary, size_t wasm_binary_size)
{
    // Parse and validate binary module.
    const FizzyModule* module = fizzy_parse(wasm_binary, wasm_binary_size);

    // Find main function.
    uint32_t main_fn_index;
    if (!fizzy_find_exported_function_index(module, "main", &main_fn_index))
        return false;

    // Instantiate module without imports.
    FizzyInstance* instance = fizzy_instantiate(module, NULL, 0, NULL, NULL, NULL, 0);

    // Execute main function without arguments.
    const FizzyExecutionResult result = fizzy_execute(instance, main_fn_index, NULL, 0);
    if (result.trapped)
        return false;

    fizzy_free_instance(instance);
    return true;
}
```

> Fizzy also provides CMake package for easy integration in the projects that use it:

Reviewer: ```suggestion
Fizzy also provides a CMake package for easy integration in projects that use it:
```

$ cmake --build .
```

This will build Fizzy as a library. [C API] is provided for embedding Fizzy engine in applications.

A small example of using the C API, which loads a wasm binary and executes a function named `main`:

```c
#include <fizzy/fizzy.h>

> bool execute_main(const uint8_t* wasm_binary, size_t wasm_binary_size)

Reviewer 2: It would be nice to create matching integration test at some point to make sure this example continue to work.

$ cmake --build .
```

> This will build Fizzy as a library. [C API] is provided for embedding Fizzy engine in applications, as well as [Rust binding].

Reviewer: I think for Rust we should give an actual example using the published crate, and perhaps link to `crates.io` and/or `docs.rs` (for documentation).

- https://crates.io/crates/fizzy
- https://docs.rs/fizzy

> This will build Fizzy as a library. [C API] is provided for embedding Fizzy engine in applications, as well as [Rust binding].

Reviewer: For C perhaps having a simple example would be useful, just to give an indication. Something which loads a wasm file and executes ta function called "main"?

Reviewer: We also have the how to include fizzy in cmake. We have a test for that.

Reviewer: @gumb0 perhaps this cmake example would be useful. See `test/integration/cmake_package/use_fizzy`.

Author: I'm not sure yet how to describe it other than mentioning `find_package(fizzy CONFIG REQUIRED)`. Can look into it later, or @chfast perhaps could help.

Author: Added a sentence anyway.

[@chfast]: https://github.com/chfast
[@gumb0]: https://github.com/gumb0

> [C API]: https://github.com/wasmx/fizzy/tree/master/include/fizzy

Reviewer: I think `./include/fizzy/fizzy.h` would work well.
